### ISSUE TITLE
compile a basic wildcard inner join to QScript

### ISSUE THREAD

**Issue by alissapajer** [quasar-analytics/quasar#1559]
```scala
    // an example of how logical plan expects magical "left" and "right" fields to exist
    "convert magical query" in {
      // "select * from person, car",
      convert(
        None,
        LP.Let('__tmp0,
          StdLib.set.InnerJoin(lpRead("/person"), lpRead("/car"), LP.Constant(Data.Bool(true))).embed,
          identity.Squash(
            structural.ObjectConcat(
              structural.ObjectProject(LP.Free('__tmp0), LP.Constant(Data.Str("left"))).embed,
              structural.ObjectProject(LP.Free('__tmp0), LP.Constant(Data.Str("right"))).embed).embed).embed)) must
      equal(chain(RootR).some) // TODO incorrect expectation
    }
```

This test is `pendingUntilFixed`.

#1556 eliminates the `"left"` and `"right"` fields, but this ticket will likely be completed before that one is resolved.

**Comment by rabbitonweb** [comment]
Copy-paste conversation with @sellout 


for
```select * from `/rabbit/hadoop/test4.txt` as left join `/rabbit/hadoop/test2.txt` as right on left.country = right.country```
I get 0 results


sellout [1:06]  
I simplified it down to this:
```Subset(
  Unreferenced,
  EquiJoin(
    Unreferenced,
      Map(ShiftedRead(/rabbit/hadoop/test4.txt, ExcludeId),
        Guard(SrcHole, Obj(Map(),Some(Top)), SrcHole, Undefined())),
      Map(ShiftedRead(/rabbit/hadoop/test2.txt, ExcludeId),
        Guard(SrcHole, Obj(Map(),Some(Top)), SrcHole, Undefined())),
      MakeArray(SrcHole),
      MakeArray(SrcHole),
      Inner,
      Guard(RightSide, Obj(Map(),Some(Top)), Guard(LeftSide, Obj(Map(),Some(Top)), ConcatMaps(LeftSide, RightSide), Undefined()), Undefined())),
  Take,
  Map(Unreferenced,Constant(Int(11))))
```

[1:06]  
Which looks right except that the keys are `MakeArray(SrcHole)` when they should be `MakeArray(ProjectField(SrcHole, StrLit("country")))`.

[1:07]  
I.e., it’s only going to return a result when both sides are equal.



Long version of generated QS

```
QScript:
List
├─ Unreferenced
╰─ Subset((),ThetaJoin(Unreferenced,LeftShift(Read(/rabbit/hadoop/test4.txt),ZipMapKeys(SrcHole),ConcatArrays(MakeArray(ConcatArrays(ConcatArrays(ConcatArrays(MakeArray(MakeMap(Constant(Str(m)), ProjectIndex(RightSide, Constant(Int(0))))), Constant(Arr(List(Fix(Coproduct(-\/(Map(List((Fix(Coproduct(\/-(Str(f)))),Fix(Coproduct(\/-(Str(test4.txt)))))))))))))), Constant(Arr(List(Fix(Coproduct(-\/(Map(List((Fix(Coproduct(\/-(Str(f)))),Fix(Coproduct(\/-(Str(hadoop)))))))))))))), Constant(Arr(List(Fix(Coproduct(-\/(Map(List((Fix(Coproduct(\/-(Str(f)))),Fix(Coproduct(\/-(Str(rabbit))))))))))))))), MakeArray(Guard(ProjectIndex(RightSide, Constant(Int(1))), Obj(Map(),Some(Top)), ProjectIndex(RightSide, Constant(Int(1))), Undefined())))),LeftShift(Read(/rabbit/hadoop/test2.txt),ZipMapKeys(SrcHole),ConcatArrays(MakeArray(ConcatArrays(ConcatArrays(ConcatArrays(MakeArray(MakeMap(Constant(Str(m)), ProjectIndex(RightSide, Constant(Int(0))))), Constant(Arr(List(Fix(Coproduct(-\/(Map(List((Fix(Coproduct(\/-(Str(f)))),Fix(Coproduct(\/-(Str(test2.txt)))))))))))))), Constant(Arr(List(Fix(Coproduct(-\/(Map(List((Fix(Coproduct(\/-(Str(f)))),Fix(Coproduct(\/-(Str(hadoop)))))))))))))), Constant(Arr(List(Fix(Coproduct(-\/(Map(List((Fix(Coproduct(\/-(Str(f)))),Fix(Coproduct(\/-(Str(rabbit))))))))))))))), MakeArray(Guard(ProjectIndex(RightSide, Constant(Int(1))), Obj(Map(),Some(Top)), ProjectIndex(RightSide, Constant(Int(1))), Undefined())))),Eq(LeftSide, RightSide),Inner,Guard(ProjectIndex(RightSide, Constant(Int(1))), Obj(Map(),Some(Top)), Guard(ProjectIndex(LeftSide, Constant(Int(1))), Obj(Map(),Some(Top)), ConcatMaps(ProjectIndex(LeftSide, Constant(Int(1))), ProjectIndex(RightSide, Constant(Int(1)))), Undefined()), Undefined())),Take,Map(Unreferenced,Constant(Int(11))))
   ╰─ ExternallyManaged(Extern)

QScript (Spark):
Subset((),Map(EquiJoin(Unreferenced,Map(ShiftedRead(/rabbit/hadoop/test4.txt, IncludeId),ConcatArrays(MakeArray(ConcatArrays(ConcatArrays(ConcatArrays(MakeArray(MakeMap(Constant(Str(m)), ProjectIndex(SrcHole, Constant(Int(0))))), Constant(Arr(List(Fix(Coproduct(-\/(Map(List((Fix(Coproduct(\/-(Str(f)))),Fix(Coproduct(\/-(Str(test4.txt)))))))))))))), Constant(Arr(List(Fix(Coproduct(-\/(Map(List((Fix(Coproduct(\/-(Str(f)))),Fix(Coproduct(\/-(Str(hadoop)))))))))))))), Constant(Arr(List(Fix(Coproduct(-\/(Map(List((Fix(Coproduct(\/-(Str(f)))),Fix(Coproduct(\/-(Str(rabbit))))))))))))))), MakeArray(Guard(ProjectIndex(SrcHole, Constant(Int(1))), Obj(Map(),Some(Top)), ProjectIndex(SrcHole, Constant(Int(1))), Undefined())))),Map(ShiftedRead(/rabbit/hadoop/test2.txt, IncludeId),ConcatArrays(MakeArray(ConcatArrays(ConcatArrays(ConcatArrays(MakeArray(MakeMap(Constant(Str(m)), ProjectIndex(SrcHole, Constant(Int(0))))), Constant(Arr(List(Fix(Coproduct(-\/(Map(List((Fix(Coproduct(\/-(Str(f)))),Fix(Coproduct(\/-(Str(test2.txt)))))))))))))), Constant(Arr(List(Fix(Coproduct(-\/(Map(List((Fix(Coproduct(\/-(Str(f)))),Fix(Coproduct(\/-(Str(hadoop)))))))))))))), Constant(Arr(List(Fix(Coproduct(-\/(Map(List((Fix(Coproduct(\/-(Str(f)))),Fix(Coproduct(\/-(Str(rabbit))))))))))))))), MakeArray(Guard(ProjectIndex(SrcHole, Constant(Int(1))), Obj(Map(),Some(Top)), ProjectIndex(SrcHole, Constant(Int(1))), Undefined())))),MakeArray(SrcHole),MakeArray(SrcHole),Inner,ConcatArrays(MakeArray(LeftSide), MakeArray(RightSide))),Guard(ProjectIndex(ProjectIndex(SrcHole, Constant(Int(1))), Constant(Int(1))), Obj(Map(),Some(Top)), Guard(ProjectIndex(ProjectIndex(SrcHole, Constant(Int(0))), Constant(Int(1))), Obj(Map(),Some(Top)), ConcatMaps(ProjectIndex(ProjectIndex(SrcHole, Constant(Int(0))), Constant(Int(1))), ProjectIndex(ProjectIndex(SrcHole, Constant(Int(1))), Constant(Int(1)))), Undefined()), Undefined())),Take,Map(Unreferenced,Constant(Int(11))))
╰─ Unreferenced
```

**Comment by jdegoes** [comment]
@rabbitonweb Are you sure this is the same issue as reported above? If not, may make sense to track this in a separate ticket.

**Comment by sellout** [comment]
Yeah, this query looks more like #1560 – notice how this issue doesn’t have `join … on`, where it’s the `on` that is broken in your example.

**Comment by jdegoes** [comment]
OK, @rabbitonweb, please open up a new one for this.

**Comment by rabbitonweb** [comment]
Done https://github.com/quasar-analytics/quasar/issues/1581

**Comment by alissapajer** [comment]
Fixed by #1592
